core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -64,6 +68,25 @@ pub struct Cast {
     pub timezone: String,
 }
 
+macro_rules! spark_cast_utf8_to_integral {
+    ($string_array:expr, $eval_mode:expr, $array_type:ty, $cast_method:ident) => {{
+        let mut cast_array = PrimitiveArray::<$array_type>::builder($string_array.len());
+        for i in 0..$string_array.len() {
```

Comment:
we probably can use iterator instead of for loop? 

and lets calc `$string_array.len()` once

Reply:
I think that `$string_array.len()` is already only computed once?

Reply:
I missed that! Thanks

Reply:
Fixed